### PR TITLE
`UploadedFile` record helper and guide

### PR DESCRIPTION
**Pull request by amitaibu**
- Explanation on how to use the `UploadedFile` (or any other name)  record
- Add a `refreshTemporaryDownloadUrl` helper function

For posterity, working PR using the helper function here https://github.com/amitaibu/ihp-landing-page/pull/18

### CODE REVIEW

File: Guide/file-storage.markdown [hunk 1]
```diff
@@ -333,6 +333,102 @@ Note that `|> buildCompany` was changed to `>>= buildCompany`.
...
+
+        -- Create the UploadedFile record. Set the signed URL, path, content-type etc.
+        uploadedFile <- newRecord @UploadedFile |> createRecord
+        uploadedFile <- uploadedFile
+            |> set #signedUrl signedUrl.url
+            |> set #signedUrlExpiredAt signedUrl.expiredAt
+            |> set #path storedFile.path
+            |> set #fileName (cs file.fileName)
+            |> set #contentType (cs $ Wai.fileContentType file)
+            |> updateRecord
```
Comment (mpscholten) [inline Guide/file-storage.markdown:394]:
```suggestion
        -- Create the UploadedFile record. Set the signed URL, path, content-type etc.
        uploadedFile <- newRecord @UploadedFile
            |> set #signedUrl signedUrl.url
            |> set #signedUrlExpiredAt signedUrl.expiredAt
            |> set #path storedFile.path
            |> set #fileName (cs file.fileName)
            |> set #contentType (cs $ Wai.fileContentType file)
            |> updateRecord
```

Any reason we can't do this in a single database operation? (I know the previous code I shared had the same issue, I just don't remember anymore why we did that, but it seems a single `createRecord` works here)

Comment (amitaibu) [reply Guide/file-storage.markdown:394]: https://github.com/amitaibu/ihp-landing-page/pull/18

Comment (mpscholten) [reply Guide/file-storage.markdown:394]: Hm, can you try to increase indentation?
```haskell
        -- Create the UploadedFile record. Set the signed URL, path, content-type etc.
        uploadedFile <- newRecord @UploadedFile
                |> set #signedUrl signedUrl.url
                |> set #signedUrlExpiredAt signedUrl.expiredAt
                |> set #path storedFile.path
                |> set #fileName (cs file.fileName)
                |> set #contentType (cs $ Wai.fileContentType file)
                |> createRecord
```

Comment (amitaibu) [reply Guide/file-storage.markdown:394]: That was it :man_facepalming:

**Review by mpscholten — APPROVED** [review]

**Pull request merged**